### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by jgarciao**
Adds a  [pull request template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository), similar to the ones used  in [ody-deployer](https://github.com/red-hat-data-services/odh-deployer/pull/206). 

When we create a PR for ods-ci the pull request body will have this content:
- [ ] Commits are squashed in a cohesive manner and have meaningful messages (see [how](https://cbea.ms/git-commit))
- [ ] Code is formatted following the _ODS-CI Robot Framework Style Guide_ (see [how](https://github.com/red-hat-data-services/ods-ci/blob/master/docs/check-code-style.md))
- [ ] For new automated tests: _Automation = Automated_ in Polarion
- [ ] For new automated tests: _Automated = Yes_ in Jira

Signed-off-by: Jorge Garcia Oncins <jgarciao@redhat.com>

### CODE REVIEW

**Comment by jgarciao** [comment]
Do not merge until #195 is merged and we have consensus about this

**Comment by lugi0** [comment]
Should we also add a check for 
`TC is authored in Polarion`
`TC has corresponding Polarion ID in its Tags`
?

**Comment by jgarciao** [comment]
- [ ] Commits are squashed in a cohesive manner and have meaningful messages (see [how](https://cbea.ms/git-commit))
- [ ] Code is formatted following the _ODS-CI Robot Framework Style Guide_ (see [how](https://github.com/red-hat-data-services/ods-ci/blob/master/docs/check-code-style.md))

For new automated tests:
- [ ] Test case is authored in Polarion
- [ ]  _Automation = Automated_ in Polarion
- [ ] _Automated = Yes_ in Jira
- [ ] Test case has tier information and Polarion ID in its Tags (more info [here](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.s819p3c5ud7p))

**Review by lugi0 — APPROVED** [review]
looks good imho

**Review by bdattoma — DISMISSED** [review]

**Comment by jgarciao** [comment]
**New proposal  (point 2 added)**

- [ ] Commits are squashed in a cohesive manner and have meaningful messages (see [how](https://cbea.ms/git-commit))
- [ ] The robocop analyser doesn't detect any issues in the test suites modified by this PR (see [check-code-style](https://github.com/red-hat-data-services/ods-ci/blob/master/docs/check-code-style.md))
- [ ] Code is formatted following the _[ODS-CI RobotFramework Style Guide](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit?usp=sharing)_

For new automated tests:
- [ ] Test case is authored in Polarion
- [ ]  _Automation = Automated_ in Polarion
- [ ] _Automated = Yes_ in Jira
- [ ] Test case has tier information and Polarion ID in its Tags (more info [here](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.s819p3c5ud7p))

File: .github/pull_request_template.md
Comment (LaVLaS) [inline .github/pull_request_template.md:9]: If/When we start to add automation for features that are in upstream ODH and won't have a polarion testcase, will not checking these boxes break any CI automation tied to this template?

Comment (jgarciao) [reply .github/pull_request_template.md:9]: It is possible that  we will set up GiHub actions to verify that robocop analyser doesn't detect any issues, but there is no automation planned about the items you mentioned (it's more like a reminder for us).

If you contribute with features present in ODH but not in RHODS we'll need a tag to indicate this so we don't run those tests.  We'll give a thought and document it in the style guide

We could do this:

- [ ] Commits are squashed in a cohesive manner and have meaningful messages (see [how](https://cbea.ms/git-commit))
- [ ] The robocop analyser doesn't detect any issues in the test suites modified by this PR (see [check-code-style](https://github.com/red-hat-data-services/ods-ci/blob/master/docs/check-code-style.md))
- [ ] Code is formatted following the _[ODS-CI RobotFramework Style Guide](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit?usp=sharing)_

For new automated tests for RHODS (not ODH):
- [ ] Test case is authored in Polarion
- [ ]  _Automation = Automated_ in Polarion
- [ ] _Automated = Yes_ in Jira
- [ ] Test case has tier information and Polarion ID in its Tags (more info [here](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.s819p3c5ud7p))

**Comment by jgarciao** [comment]
Latest version:

- [ ] Commits are squashed in a cohesive manner and have meaningful messages (see [how](https://cbea.ms/git-commit))
- [ ] The robocop analyser doesn't detect any issues in the test suites modified by this PR (see [check-code-style](https://github.com/red-hat-data-services/ods-ci/blob/master/docs/check-code-style.md))
- [ ] Code is formatted following the _[ODS-CI RobotFramework Style Guide](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit?usp=sharing)_
- [ ] Jira Link(s):
- [ ] An entry has been added to the corresponding _RHODS build announcement_ (see [folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL))

For new automated tests for RHODS (not ODH):
- [ ] Test case is authored in Polarion
- [ ]  _Automation = Automated_ in Polarion
- [ ] _Automated = Yes_ in Jira
- [ ] Test case contains Polarion ID, tier information and all mandatory Tags (more info [here](https://docs.google.com/document/d/11ZJOPI1uq-0Wl6a2V8fkAv_TQhfzp9t_IjXAheaJxmQ/edit#heading=h.s819p3c5ud7p))

**Review by lugi0 — APPROVED** [review]
lgtm

**Comment by bdattoma** [comment]
@jgarciao I'm closing this one, we haven't talked about the need of a template for long time. Feel free to add the topic in the agenda and we can work on this again in the near future

**Pull request closed without merging**
